libioc/Jail.py

```diff
@@ -1653,93 +1654,70 @@ def devfs_ruleset(self) -> libioc.DevfsRules.DevfsRuleset:
             ruleset_line_position = self.host.devfs.index(devfs_ruleset)
             return self.host.devfs[ruleset_line_position].number
 
-    @property
-    def _launch_command(self) -> typing.List[str]:
-
-        command = ["/usr/sbin/jail", "-c"]
+    @staticmethod
+    def __get_launch_command(jail_args: typing.List[str]) -> typing.List[str]:
+        return ["/usr/sbin/jail", "-c"] + jail_args
 
-        if self.config["vnet"]:
-            command.append("vnet")
-        else:
+    @property
+    def _launch_args(self) -> typing.List[str]:
+        config = self.config
+        value: str
+        jail_param_args: typing.List[str] = []
+        for sysctl_name, sysctl in libioc.JailParams.JailParams().items():
+            if sysctl.ctl_type == freebsd_sysctl.types.NODE:
+                # skip NODE
+                continue
 
-            if self.config["ip4_addr"] is not None:
-                ip4_addr = self.config["ip4_addr"]
-                command += [
-                    f"ip4.addr={ip4_addr}",
-                    f"ip4.saddrsel={self.config['ip4_saddrsel']}",
-                    f"ip4={self.config['ip4']}",
-                ]
+            vnet = (config["vnet"] is True)
+
+            if sysctl_name == "security.jail.param.devfs_ruleset":
+                value = str(self.devfs_ruleset)
+            elif sysctl_name == "security.jail.param.path":
+                value = self.root_dataset.mountpoint
+            elif sysctl_name == "security.jail.param.name":
+                value = self.identifier
+            elif sysctl_name == "security.jail.param.allow.mount.zfs":
+                value = str(self._allow_mount_zfs)
+            elif sysctl_name == "security.jail.vnet":
```

Comment:
what's the difference between `security.jail.param.*` and `security.jail.*`?

Reply:
Some keys are duplicated in `security.jail.*` and `security.jail.param.*`, but we only handle the latter. Fixed by rebasing.